src/bin/cargo/cli.rs

```diff
@@ -55,6 +55,13 @@ pub fn main(config: &mut Config) -> CliResult {
         }
     };
 
```

Comment:
> Because a struct Config gets created before command line arguments are processed, a config will exist with the actual cwd recorded, and it must then be replaced with the new value after command line arguments are processed but before anything tries to use the stored cwd value or any other value derived from it for anything. This change effectively creates a difficult-to-document requirement during cargo initialization regarding the order of events. For example, should a setting stored in a config file discovered via cwd+ancestors search be wanted before argument processing happens, this could result in unpleasant surprises in the exact use case this feature is being added to fix.

Github won't let me comment on the right line but there is a comment similar to this already at the start of this function